src/main/java/patching/Diff.java

```diff
@@ -175,52 +175,25 @@ public Diff getUndo() {
 
         for (Diff other : others) {
             List<Diff> newIntermediateDiffs = new ArrayList<>();
-            for (Diff intermediate : intermediateDiffs) {
+            for (Diff current : intermediateDiffs) {
 
                 // CASE 1: IndexA < IndexB
-                if (other.startIndex < intermediate.startIndex) {
+                if (other.startIndex < current.startIndex) {
                     // CASE 1a: Ins - Ins
-                    if (other.insertion && intermediate.insertion) {
-                        int newStartLoc = intermediate.startIndex + other.getLength();
-                        Diff newDiff = new Diff(intermediate.insertion, newStartLoc, intermediate.changes);
-                        newIntermediateDiffs.add(newDiff);
+                    if (other.insertion && current.insertion) {
+                        transformType2(newIntermediateDiffs, current, other);
                     }
                     // CASE 1b: Ins - Rmv
-                    else if (other.insertion && !intermediate.insertion) {
-                        int newStartLoc = intermediate.startIndex + other.getLength();
-                        Diff newDiff = new Diff(intermediate.insertion, newStartLoc, intermediate.changes);
-                        newIntermediateDiffs.add(newDiff);
-
+                    else if (other.insertion && !current.insertion) {
```

Comment:
Yeah. It's more verbose, but it's also a little more specific. It's hard enough tracing the logic and matching it to the OT effects excel sheet.